test/hotspot/jtreg/compiler/vectorapi/VectorFPtoIntCastTest.java

```diff
@@ -0,0 +1,221 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @bug 8287835
+* @summary Test float/double to integral cast
+* @modules jdk.incubator.vector
+* @requires vm.compiler2.enabled
+* @requires (os.simpleArch == "x64" & vm.cpu.features ~= ".*avx512dq.*")
+* @library /test/lib /
+* @run driver compiler.vectorapi.VectorFPtoIntCastTest
+*/
+
+package compiler.vectorapi;
+
+import jdk.incubator.vector.*;
+import jdk.incubator.vector.FloatVector;
+import compiler.lib.ir_framework.*;
+import java.util.Random;
+
+public class VectorFPtoIntCastTest {
+    private static final int COUNT = 16;
+    private static final VectorSpecies<Long> lspec512 = LongVector.SPECIES_512;
+    private static final VectorSpecies<Integer> ispec512 = IntVector.SPECIES_512;
+    private static final VectorSpecies<Integer> ispec256 = IntVector.SPECIES_256;
+    private static final VectorSpecies<Short> sspec256 = ShortVector.SPECIES_256;
+    private static final VectorSpecies<Short> sspec128 = ShortVector.SPECIES_128;
+    private static final VectorSpecies<Byte> bspec128 = ByteVector.SPECIES_128;
+    private static final VectorSpecies<Byte> bspec64  = ByteVector.SPECIES_64;
+
+    private float [] float_arr;
+    private double [] double_arr;
+    private long [] long_arr;
+    private int [] int_arr;
+    private short [] short_arr;
+    private byte [] byte_arr;
+
+    private FloatVector fvec256;
+    private FloatVector fvec512;
+    private DoubleVector dvec512;
+
+    public static void main(String args[]) {
+        TestFramework.runWithFlags("--add-modules=jdk.incubator.vector");
+    }
+
+    public VectorFPtoIntCastTest() {
+        float_arr = new float[COUNT];
+        double_arr = new double[COUNT];
+        long_arr = new long[COUNT];
+        int_arr = new int[COUNT];
+        short_arr = new short[COUNT];
+        byte_arr = new byte[COUNT];
+
+        Random ran = new Random(0);
+        for (int i = 0; i < COUNT; i++) {
+            float_arr[i] = ran.nextFloat();
+            double_arr[i] = ran.nextDouble();
+        }
```

Comment:
Can you kindly all add special floating point cases NaN, +/-Inf, +/-0.0 to cover your special handling code changes.

Reply:
@jatin-bhateja The test is only checking the IR node generation for x86. 
The rest of the actual functionality test is already covered under the following including the special cases:
compiler/codegen/TestByteDoubleVect.java compiler/codegen/TestByteFloatVect.java compiler/codegen/TestShortFloatVect.java compiler/codegen/TestShortDoubleVect.java compiler/codegen/TestLongFloatVect.java compiler/codegen/TestIntDoubleVect.java compiler/codegen/TestIntFloatVect.java
The general idea of this PR was to complement x86 FP to integral conversion along with https://git.openjdk.org/jdk/pull/7806 from Fei Gao.